discovery_engine_core/core/src/stack/filters.rs

```diff
@@ -12,23 +12,51 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use crate::{document::Document, engine::GenericError};
 use std::collections::HashSet;
+
 use url::Url;
+
+use crate::{
+    document::{Document, HistoricDocument},
+    engine::GenericError,
+};
 use xayn_discovery_engine_providers::Article;
 
 pub(crate) trait ArticleFilter {
-    fn apply(current: &[Document], articles: Vec<Article>) -> Result<Vec<Article>, GenericError>;
+    fn apply(
+        history: &[HistoricDocument],
+        stack: &[Document],
+        articles: Vec<Article>,
+    ) -> Result<Vec<Article>, GenericError>;
+}
+
+struct HistoryFilter;
+
+impl ArticleFilter for HistoryFilter {
```

Comment:
Can't we do this already in `Duplicatefilter`?

Reply:
sure we can. my impression of the previous design was that all filters should be separate such that they can be combined in the `CommonFilter` later on. but we can also combine the `HistoryFilter` and the `DuplicateFilter` into one, i don't mind much.